roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -0,0 +1,121 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class DiskAvailability(OpenShiftCheck):
+    """Check that recommended disk space is available."""
+
+    name = "disk_availability"
+    tags = ["preflight"]
+
+    recommended_diskspace = {
+        "nodes": 15*10**9,
+        "masters": 40*10**9,
+        "etcd": 20*10**9,
+    }
+
+    minimum_docker_space_percent = 5.0
+    supported_docker_dev_driver = "devicemapper"
+
+    def run(self, tmp, task_vars):
+        self.minimum_docker_space_percent = get_var(task_vars,
+                                                    "healthcheck_minimum_docker_diskspace_percent",
+                                                    default=self.minimum_docker_space_percent)
+
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+
+        if is_containerized:
+            failed, msg = self.containerized_volume_check(task_vars)
+            if failed:
+                return {"failed": True, "msg": msg}
+
+            return {"changed": False}
+
+        failed, msg = self.noncontainerized_volume_check(ansible_mounts, task_vars)
+        return {"failed": failed, "msg": msg}
+
+    def containerized_volume_check(self, task_vars):
```

Comment:
Thanks, per the comments on  https://github.com/openshift/openshift-ansible/pull/3518#discussion_r110243468, I have decided to leave all aspects of the containerized check to https://github.com/openshift/openshift-ansible/pull/3787. (I will further refine that PR, rename it so the name is consistent with the check in this PR, and split out the storage driver check from it into its own check). Any of your feedback on #3787 is definitely welcome :)